src/WebJobs.Extensions/Extensions/Timers/Scheduling/CronSchedule.cs

```diff
@@ -32,13 +51,103 @@ internal CrontabSchedule InnerSchedule
+            //
+            // For "interval" schedules, we want to continue running through this ambiguous hour as usual. Using an "every 30 minute" schedule
+            // and Pacific time offset an example, we'd want to:
+            //    - Run at 01:00-7, 01:30-7, 01:00-8, 01:30-8, 02:00-8, 02:30-8, etc.
+            //
+            // For "point-in-time" schedules, we only want to run them once (i.e a 1:30 trigger only runs once on this day). Using an
+            // "every day at 01:30" schedule and Pacific time offset as an example in 2018, we'd want to:
+            //    - Run on November 3rd at 01:30-7
+            //    - Run on November 4th at 01:30-7 (and not run at 1:30-8)
+            //    - Run on November 5th at 01:30-8
+            if (!IsInterval && isNextAmbiguous && nextOffset != now.Offset)
+            {
+                // Scenario 2 -- Point-in-time schedule where the next time is ambiguous and the offsets have changed.
+                //   This means that "now" is in DST and "next" is in Standard Time. Since we never want to run an ambiguous
+                //   point-in-time schedule on Standard time, move the offset back to the Daylight offset.
+                var offsetDiff = nextOffset - now.Offset;
+                next = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), TimeZone);
+            }
+            else if (!IsInterval && isNextAmbiguous && nextOffset == now.Offset)
+            {
+                // Scenario 3 -- Point-in-time where "next" is ambiguous and "now" and "next" are in Standard time. This can happen
+                //   when the timer starts and we are already past the "fall back" point. For example, if the trigger starts up at
+                //   01:29-8 and we have a "every day at 01:30" schedule, we have to assume that we've already run it at 01:30-7 and
+                //   therefore calculate the time after this.
+                while (TimeZone.IsAmbiguousTime(nextDateTime))
+                {
+                    nextDateTime = _cronSchedule.GetNextOccurrence(nextDateTime);
+                }
+
+                next = new DateTimeOffset(nextDateTime, TimeZone.GetUtcOffset(nextDateTime));
+            }
+            else if (IsInterval && (isNowAmbiguous || isNextAmbiguous) && nextOffset != now.Offset)
+            {
+                // Scenario 4 -- Interval schedule where either point is ambiguous and we're crossing the offset boundary. For example, 
+                //   an "every 30 minute" interval in Pacific time:
+                //   - 00:30-7 -> 01:00-8 (because all ambiguous "next" use Standard time) -> adjust back 01:00-7
+                //   - 01:30-7 -> 02:00-8 -> adjust back to 01:00-8
+                var offsetDiff = nextOffset - now.Offset;
+                next = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), TimeZone);
+            }
+
+            return next;
```

Comment:
I think this all mostly makes sense to me; I'm just struggling to wrap my head around the difference in the "spring forward" vs "fall back" calculations since they're all happening in this one function i.e. we're checking for DST in both directions every time which is harder to follow for me

Reply:
Yes, I agree. Let me break them out to make this more clear. I realized that as I was reading through it again also.